app/api/ada/hardwareWallet/newTransaction.js

```diff
@@ -154,3 +159,90 @@ function _outputAddressOrPath(out: TxOutput) {
   }
   return { address: out.address };
 }
+
+/** Generate a payload for Ledger SignTx */
+export async function createLedgerSignTxPayload(
+  txExt: UnsignedTransactionExt
+): Promise<LedgerSignTxPayload> {
+
+  // Transactions Hash
+  const txDataHexList = await txsBodiesForInputs(txExt.inputs);
+
+  // Inputs
+  const ledgerInputs: Array<LedgerInputTypeUTxO> =
+    _transformToLedgerInputs(txExt.inputs, txDataHexList);
+
+  // Outputs
+  const ledgerOutputs = _transformToLedgerOutputs(txExt.outputs);
+
+  return {
+    inputs: ledgerInputs,
+    outputs: ledgerOutputs,
+  };
+}
+
+function _transformToLedgerInputs(
+  inputs: Array<TxInput>, 
+  txDataHexList: Array<string>
+): Array<LedgerInputTypeUTxO> {
+  return inputs.map((input: TxInput, idx: number) => ({
+    txDataHex: txDataHexList[idx],
+    outputIndex: input.ptr.index,
+    path: _derivePath(input.addressing.change, input.addressing.index),
+  }));
+}
+
+function _transformToLedgerOutputs(
+  outputs: Array<TxOutput>
+): Array<LedgerOutputTypeAddress | LedgerOutputTypeChange> {
+  return outputs.map(x => ({
+    amountStr: x.value.toString(),
+    ..._ledgerOutputAddressOrPath(x)
+  }));
+}
+
+function _ledgerOutputAddressOrPath(out: TxOutput) {
```

Comment:
@sarveshwar-singh missing return type and out is not a great name. This would be better with a type that is basically a string but it’s called Base58Address or something like that

Reply:
@nicarq 
As per my understanding there are 3 request:
1. missing return type
2. change the name to function parameters
3. [ Base58Address, is this about return type ?

For 1, 2 should i also change original functions which was written for Trezor Sign Tx part?

Reply:
If you add it, now you own it lol 😄